### PR TITLE
Set up Initial project

### LINKED ISSUE #1: Set up Initial project
## Why

To start project

## Acceptance Criteria
- [x]  Initialize the project with Beego framework 
- [x]  Support Docker
- [x] Support Postgres
- [x] Support 
- [x] Set up Webpack
- [x] Set up Boostrap
- [x] Set up scss
- [x] Project able to run

### PR DESCRIPTION
**Pull request by junan**
https://github.com/junan/google-scraper/projects/1#card-50537698

## What happened 👀

Initialized the Beego project
 
## Insight 📝

- Generated the project with `bee new` command
- Used Beego 2 version
- dockerized the application

**Note:** Database, ORM, style integration will be done in [4](https://github.com/junan/google-scraper/issues/4)

## Proof Of Work 📹
Screen Shot 2020-12-17 at 4 14 28 PM

### CODE REVIEW

File: tests/default_test.go [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+	w := httptest.NewRecorder()
+	beego.BeeApp.Handlers.ServeHTTP(w, r)
+
+	log.Trace("testing", "TestBeego", "Code[%d]\n%s", w.Code, w.Body.String())
+
+	Convey("Subject: Test Station Endpoint\n", t, func() {
+		Convey("Status Code Should Be 200", func() {
+			So(w.Code, ShouldEqual, 200)
+		})
+		Convey("The Result Should Not Be Empty", func() {
```
Comment (olivierobert) [inline tests/default_test.go]: Prefer the imperative tone and using lower case in tests:

```suggestion
		Convey("The result is not empty", func() {
```

**Review by olivierobert — CHANGES_REQUESTED** [review]
The PR title must match the related task:

image


So it should be "Set up Initial project"

File: tests/default_test.go (hunk 1 above)
Comment (junan) [reply tests/default_test.go]: Updated, these are actually generated by `bee` command

Comment (olivierobert) [reply tests/default_test.go]: Yes but you should also check those :-)

**Review by olivierobert — APPROVED** [review]

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,13 @@
+.PHONY: test dev production
+
+build-dependencies:
```
Comment (gutakk) [inline Makefile:3]: Should `build-dependencies` be added to `.PHONY`?

File: go.mod [hunk 3]
```diff
@@ -0,0 +1,9 @@
+module google-scraper
+
+go 1.15
+
+require (
+	github.com/beego/bee/v2 v2.0.2 // indirect
+	github.com/beego/beego/v2 v2.0.1
```
Comment (gutakk) [inline go.mod:7]: Why do we have 2 versions of beego here? Can we remove one of them out?

File: main.go [hunk 4]
```diff
@@ -0,0 +1,10 @@
+package main
+
+import (
+	_ "google-scraper/routers"
+	beego "github.com/beego/beego/v2/server/web"
```
Comment (gutakk) [inline main.go:6]: How about grouping the import follow our [Go conventions](https://nimblehq.co/compass/development/code-conventions/golang/#imports)?

File: Makefile (hunk 2 above)
Comment (junan) [reply Makefile:3]: @gutakk Done

File: main.go (hunk 4 above)
Comment (junan) [reply main.go:6]: Added extra line between them

File: go.mod (hunk 3 above)
Comment (junan) [reply go.mod:7]: @gutakk Its actually one for `bee`(CLI tool for generate the `beego` project), other one is for `beego` itself. These are auto generated.

Comment (gutakk) [reply go.mod:7]: Thanks for clarify.

**Review by gutakk — APPROVED** [review]

**Review by carryall — APPROVED** [review]

**Pull request merged**